### ISSUE TITLE
[EA] Null pointer exception may occur here. False positive with nullsafe operator. 

### ISSUE THREAD

**Issue by lysiakin** [kalessil/phpinspectionsea#1826]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.7.1 |
| Language level | e.g. PHP 8.1 (check project settings or composer.json)                        |

Current behaviour
--
Using nullable 'getLast' within 'if' with nullsafe operator EA warns about 'null pointer exception'. 

![image](https://user-images.githubusercontent.com/45404239/164974333-a42f9058-5c1b-4078-a76d-e5dbe8f42a37.png)
![image](https://user-images.githubusercontent.com/45404239/164974455-2b7eb6a7-3fe7-4f0b-86df-29e6a9579452.png)

Expected behaviour
--

When using nullable 'getLast' within 'if' with nullsafe operator EA shouldn't warn about 'null pointer exception'. 
Environment details

---

Information from Help - About dialog

**Comment by rentalhost** [comment]
I think it's an imposition of the IDE itself, and it makes perfect sense. Well, in your case, it doesn't, since we have the notion that the code will always return the same information that has already been validated. But imagine that your function might have some conditionals where the second call might respond differently from the first.

In this case, my suggestion is to store the validated information in a variable:

```php
$queryName = $query->getName();

if ($queryName?->getLast()) {
    $qb
        ->andWhere('v.name.last = :tier')
        ->setParameter('name', $queryName->getLast())
    ;
}
```

Or:

```php
$queryNameLast = $query->getName()?->getLast();

if ($queryNameLast) {
    $qb
        ->andWhere('v.name.last = :tier')
        ->setParameter('name', $queryNameLast)
    ;
}
```
